From 4efd50415449a6063dc513a8dcd2b732b077c773 Mon Sep 17 00:00:00 2001
From: Ken Wahid <kenx.wahid@intel.com>
Date: Fri, 17 Feb 2012 19:02:33 +0100
Subject: [PORT FROM MAIN] Make the PCSC-lite daemon run as a system service.

BZ: 24267

DO NOT UPSTREAM

Changed the PSCS-lite daemon to run as an independent system service
instead of as a child process of the wpa_supplicant service.
This is to allow the PCSC-lite daemon to perform proper cleanup at
termination.
See the comment in bugzilla for further implementation details.

Change-Id: Ibdf6d11e9acb7f5e9a2ce734d7c867621bcdbaf0
Orig-Change-Id: I5cba6a0e6fa79bca69b9bfa422e6c749d316c692
Signed-off-by: Ken Wahid <kenx.wahid@intel.com>
Reviewed-on: http://android.intel.com:8080/35789
Reviewed-by: Niciarz, Sebastian <sebastian.niciarz@intel.com>
Reviewed-by: Yin, Fengwei <fengwei.yin@intel.com>
Tested-by: Zurmely, PierreX <pierrex.zurmely@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Signed-off-by: Sebastian Niciarz <sebastian.niciarz@intel.com>
Reviewed-on: http://android.intel.com:8080/60234
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-on: http://android.intel.com:8080/75497
Reviewed-by: lab_cactus <lab_cactus@intel.com>
Reviewed-by: Rebraca, DejanX <dejanx.rebraca@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 init/property_service.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/init/property_service.c b/init/property_service.c
index 62b6c3d..384a414 100755
--- a/init/property_service.c
+++ b/init/property_service.c
@@ -107,6 +107,7 @@ struct {
 } control_perms[] = {
     { "dumpstate",AID_SHELL, AID_LOG },
     { "ril-daemon",AID_RADIO, AID_RADIO },
+    { "pcsc",AID_WIFI, AID_WIFI },  /* Allow wpa_supplicant to start the pcsc-lite daemon used for EAP-SIM/AKA auth */
      {NULL, 0, 0 }
 };
 
-- 
1.7.4.1

